src/lang/wasm.ts

```diff
@@ -136,27 +141,44 @@ export const parse = (code: string | Error): Program | Error => {
 
 export type PathToNode = [string | number, string][]
 
-interface Memory {
-  [key: string]: KclValue
+export interface ExecState {
+  memory: ProgramMemory
+  idGenerator: IdGenerator
 }
 
-type EnvironmentRef = number
+/**
+ * Create an empty ExecState.  This is useful on init to prevent needing an
+ * Option.
+ */
+export function emptyExecState(): ExecState {
+  return {
+    memory: ProgramMemory.empty(),
+    idGenerator: defaultIdGenerator(),
+  }
+}
 
-const ROOT_ENVIRONMENT_REF: EnvironmentRef = 0
+function execStateFromRaw(raw: RawExecState): ExecState {
+  return {
+    memory: ProgramMemory.fromRaw(raw.memory),
+    idGenerator: raw.idGenerator,
+  }
+}
 
-interface Environment {
```

Comment:
I didn't touch the Environment type in this PR. It's auto generated now by ts_rs, so some code actually got deleted. So I think git just couldn't handle it.

Reply:
The only significant change here is that the values in Memory could be undefined now. Nothing actually changed, but the types are generated by ts_rs instead of the hand written ones I made before.